torch/csrc/jit/codegen/cuda/dispatch.h

```diff
@@ -61,12 +61,11 @@ class IterDomain;
 class TensorDomain;
 class TensorView;
 
-class Bool;
-template <DataType DT>
-class FloatingPoint;
-using Float = FloatingPoint<DataType::Float>;
-using Double = FloatingPoint<DataType::Double>;
-class Int;
+template <typename DT>
+class Scalar;
+using Bool = Scalar<bool>;
+using Double = Scalar<double>;
+using Int = Scalar<int64_t>;
```

Comment:
This is a bit confusing. Should we change the name to `FloatingPoint` and `Integer`?

Reply:
I'm fine with the names, just didn't want to change the existing names. I'd prefer shorter nameas and keep `Int` as is and rename `Double` to `Float`. What do you think?

Reply:
That sounds good to me

Reply:
Actually, `Float` seems a little off to me as the default underlying type is `double`, `Double` seems more intuitive. Let's keep the names as is.